File: Helpers/src/main/java/com/dg/helpers/DismissSoftkeyboardHelper.java
package com.dg.helpers;

import android.app.Activity;
import android.content.Context;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;
import android.view.inputmethod.InputMethodManager;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.Checkable;
import android.widget.DatePicker;
import android.widget.EditText;
import android.widget.ImageButton;

/**
 * Created by Daniel Cohen Gindi (danielgindi@gmail.com)
 */
public class DismissSoftkeyboardHelper
{

    public static void setupUI(final Context context, View view)
    {
        if (view == null) return;

        if (!(view instanceof EditText) &&
                !(view instanceof Button) &&
                !(view instanceof ImageButton) &&
                !(view instanceof Checkable) &&
                !(view instanceof DatePicker) &&
                !(view instanceof AdapterView) &&
                !(view instanceof android.widget.NumberPicker) &&
                !(view instanceof android.widget.RadioGroup) &&
                !(view instanceof android.widget.TimePicker) &&
                !(view instanceof android.widget.Toolbar))
        {
            view.setOnTouchListener(new View.OnTouchListener()
            {
                public boolean onTouch(View v, MotionEvent event)
                {
                    hideSoftKeyboard(context);
                    return false;
                }
            });
        }

        if (view instanceof ViewGroup)
        {
            for (int i = 0; i < ((ViewGroup) view).getChildCount(); i++)
            {
                View innerView = ((ViewGroup) view).getChildAt(i);
                setupUI(context, innerView);
            }
        }
    }

    public static void hideSoftKeyboard(Context context)
    {
        try
        {
            InputMethodManager inputMethodManager = (InputMethodManager) context.getSystemService(Context.INPUT_METHOD_SERVICE);
            inputMethodManager.hideSoftInputFromWindow(((Activity) context).getCurrentFocus().getWindowToken(), 0);
        }
        catch (Exception ex)
        {

        }
    }
}

File: Helpers/src/main/java/com/dg/helpers/DismissSoftkeyboardHelper.kt
package com.dg.helpers

import android.app.Activity
import android.content.Context
import android.os.Build
import android.view.View
import android.view.ViewGroup
import android.view.inputmethod.InputMethodManager
import android.widget.*

@Suppress("unused")
object DismissSoftkeyboardHelper
{
    @Suppress("MemberVisibilityCanBePrivate")
    fun setupUI(context: Context, view: View?)
    {
        if (view == null) return

        val viewPackageName = view.javaClass.getPackage()?.name
        val viewClassName = view.javaClass.simpleName

        if (view is EditText ||
                view is Button ||
                view is ImageButton ||
                view is Checkable ||
                view is DatePicker ||
                view is AdapterView<*> ||
                view is NumberPicker ||
                view is RadioGroup ||
                view is TimePicker ||
                Build.VERSION.SDK_INT >= 21 && view is Toolbar ||
                viewPackageName == "android.widget" && viewClassName == "Toolbar" ||
                viewClassName == "TextInputLayout")
        {
            return
        }

        view.setOnTouchListener { _, _ ->
            hideSoftKeyboard(context)
            false
        }

        if (view is ViewGroup)
        {
            for (i in 0 until view.childCount)
            {
                val innerView = view.getChildAt(i)
                setupUI(context, innerView)
            }
        }
    }

    @Suppress("MemberVisibilityCanBePrivate")
    fun hideSoftKeyboard(context: Context)
    {
        try
        {
            val inputMethodManager = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            inputMethodManager.hideSoftInputFromWindow((context as Activity).currentFocus!!.windowToken, 0)
        }
        catch (ex: Exception)
        {

        }

    }
}